File: .github/scripts/update-prs.js
const fs = require("fs");

const SECTION_HEADER = "## 🛠️ Latest Open Source Work";
const USERNAME = "ArjunJagdale";
const REPO = "huggingface/datasets";

(async () => {
  const fetch = (await import("node-fetch")).default;

  const res = await fetch(`https://api.github.com/search/issues?q=type:pr+author:${USERNAME}+repo:${REPO}+is:public&sort=created&order=desc`);
  const data = await res.json();

  const items = data.items?.slice(0, 5) || [];

  const markdown = items
    .map(pr => `- [${pr.title} (#${pr.number})](${pr.html_url})`)
    .join("\n");

  const readme = fs.readFileSync("README.md", "utf-8");

  const updated = readme.replace(
    new RegExp(`${SECTION_HEADER}[\\s\\S]*?(?=\\n##|\\n---|\\n$)`, "g"),
    `${SECTION_HEADER}\n\n${markdown}`
  );

  fs.writeFileSync("README.md", updated);
})();

File: .github/scripts/update_prs.py
import requests
import os
import re

GH_USERNAME = os.getenv("GH_USERNAME", "ArjunJagdale")
GH_TOKEN = os.getenv("GH_PAT")  # ✅ Match your GitHub Actions secret

headers = {"Authorization": f"token {GH_TOKEN}"} if GH_TOKEN else {}

url = f"https://api.github.com/search/issues?q=type:pr+author:{GH_USERNAME}+repo:huggingface/datasets+is:public&sort=created&order=desc"
response = requests.get(url, headers=headers)
data = response.json()
items = data.get("items", [])[:5]

markdown_lines = [f"- [{item['title']} (#{item['number']})]({item['html_url']})" for item in items]
markdown = "\n".join(markdown_lines)

with open("README.md", "r", encoding="utf-8") as f:
    readme = f.read()

pattern = r"(## 🛠️ Latest Open Source Work\n\n)([\s\S]*?)(?=\n##|\n---|\n$)"
new_readme = re.sub(pattern, rf"\1{markdown}", readme)

with open("README.md", "w", encoding="utf-8") as f:
    f.write(new_readme)
